src/library_pthread.js

```diff
@@ -159,11 +162,11 @@ var LibraryPThread = {
 #if PTHREADS_DEBUG
       err('terminateAllThreads');
 #endif
-      for (var t in PThread.pthreads) {
-        var worker = PThread.pthreads[t];
-        if (worker) {
-          PThread.returnWorkerToPool(worker);
-        }
+      for (var worker of Object.values(PThread.pthreads)) {
+#if ASSERTIONS
+        assert(worker);
```

Comment:
Why is this assert guaranteed to never fire? The code before doesn't seem to assume that invariant.

Reply:
I think the old code was just being needlessly conservative here.  Maybe at some point in the past it was possible to put something other than a worker into the `pthreads` map.. but its clear from the code today that this map should contain workers and only workers.. never `null` and never `undefined`, just valid workers. 

I added the assertion just to be extra careful that I'm right about that.  All the tests still pass and we have pretty great coverage of pthreads code these days (especially with the posixtest suite).

Reply:
I can split this change off maybe?

Reply:
I see, thanks. I thought this was just a refactoring of the code, and I didn't get why it was equivalent. But if this is a new assertion that sounds ok to me too.